wp-cli-rename-db-prefix.php

```diff
@@ -28,292 +28,290 @@
-	 *
-	 * @throws Exception
-	 */
-	protected function update_wp_config() {
-		if ( $this->is_dry_run ) {
-			return;
-		}
-
-		if ( ! $this->is_config_update ) {
-			\WP_CLI::line( 'Skipping wp-config.php update as requested.' );
-			return;
-		}
-
-		$wp_config_path     = \WP_CLI\Utils\locate_wp_config(); // we know this is valid, because wp-cli won't run if it's not
-		$wp_config_contents = file_get_contents( $wp_config_path );
-		$search_pattern     = '/(\$table_prefix\s*=\s*)([\'"]).+?\\2(\s*;)/';
-		$replace_pattern    = "\${1}'{$this->new_prefix}'\${3}";
-		$wp_config_contents = preg_replace( $search_pattern, $replace_pattern, $wp_config_contents, - 1, $number_replacements );
-
-		if ( 0 === $number_replacements ) {
-			throw new Exception( "Failed to replace `\$table_prefix` in `wp-config.php`." );
-		}
-
-		if ( ! file_put_contents( $wp_config_path, $wp_config_contents ) ) {
-			throw new Exception( "Failed to update updated `wp-config.php` file." );
-		}
-	}
-
-	/**
-	 * Rename all of WordPress' database tables
-	 *
-	 * @throws Exception
-	 */
-	protected function rename_wordpress_tables() {
-		global $wpdb;
-
-		$show_table_query = sprintf(
-			'SHOW TABLES LIKE "%s%%";',
-			$wpdb->esc_like( $this->old_prefix )
-		);
-
-		$tables = $wpdb->get_results( $show_table_query, ARRAY_N );
-
-		if ( ! $tables ) {
-			throw new Exception( 'MySQL error: ' . $wpdb->last_error );
-		}
-
-		foreach ( $tables as $table ) {
-			$table = substr( $table[0], strlen( $this->old_prefix ) );
-
-			$rename_query = sprintf(
-				"RENAME TABLE `%s` TO `%s`;",
-				$this->old_prefix . $table,
-				$this->new_prefix . $table
-			);
-
-			if ( $this->is_dry_run ) {
-				\WP_CLI::line( $rename_query );
-				continue;
-			}
-
-			if ( false === $wpdb->query( $rename_query ) ) {
-				throw new Exception( 'MySQL error: ' . $wpdb->last_error );
-			}
-		}
-	}
-
-	/**
-	 * Update rows in all of the site `options` tables
-	 *
-	 * @throws Exception
-	 */
-	protected function update_blog_options_tables() {
-		global $wpdb;
-
-		if ( ! is_multisite() ) {
-			return;
-		}
-
-		throw new Exception( 'Not done yet' );
-
-		// todo this hasn't been tested at all
-		// todo should this really go after update_options_table, and reuse the same query?
-		// todo is this running on the root site twice b/c update_options_table() hits that too? should call either that or this, based on is_multisite() ?
-
-		$sites = get_sites( array( 'number' => false ) );
-		//blogs = $wpdb->get_col( "SELECT blog_id FROM `" . $this->new_prefix . "blogs` WHERE public = '1' AND archived = '0' AND mature = '0' AND spam = '0' ORDER BY blog_id DESC" );
-
-		if ( ! $sites ) {
-			throw new Exception( 'Failed to get all sites.' );  // todo test
-		}
-
-		foreach ( $sites as $site ) {
-			$update_query = $wpdb->prepare( "
+    public $old_prefix;
+    public $new_prefix;
+
+    public $is_dry_run = false;
+    public $is_prompt = true;
+    public $is_config_update = true;
+
```

Comment:
You'll need to restore all the original spacing, etc, to remove all this diff noise. 

It's difficult to see what the changes are until that's done. The only modified lines in the diff should be the ones that you intend to change.

Reply:
GitHub has a cog wheel up there: https://github.com/iandunn/wp-cli-rename-db-prefix/pull/17/files?diff=unified&w=1